src/plugins/index_pattern_field_editor/public/components/preview/documents_nav_preview.tsx

```diff
@@ -0,0 +1,156 @@
+          {
+            defaultMessage: 'Document not found',
+          }
+        )
+      : null;
+  const isInvalid = error !== null;
+
+  // We don't display the nav button when the user has entered a custom
+  // document ID as at that point there is no more reference to what's "next"
+  const showNavButtons = isCustomID === false;
+
+  const onDocumentIdChange = useCallback((e: React.SyntheticEvent<HTMLInputElement>) => {
+    setIsCustomID(true);
+    const nextId = e.currentTarget.value;
+    setDocumentId(nextId);
+  }, []);
+
+  const loadDocFromCluster = useCallback(() => {
+    lastDocumentLoaded.current = null;
+    setIsCustomID(false);
+    loadFromCluster();
+  }, [loadFromCluster]);
+
+  useEffect(() => {
+    if (currentDocument && !isCustomID) {
+      setDocumentId(currentDocument._id);
+    }
+  }, [currentDocument, isCustomID]);
+
+  useDebounce(
+    () => {
+      if (!isCustomID || !Boolean(documentId.trim())) {
+        return;
+      }
+
+      if (lastDocumentLoaded.current === documentId) {
+        return;
+      }
+
+      lastDocumentLoaded.current = documentId;
+
+      loadSingle(documentId);
+    },
+    500,
+    [documentId, isCustomID]
+  );
+
+  return (
+    <div>
+      <EuiFlexGroup gutterSize="s">
+        <EuiFlexItem>
+          <EuiFormRow
+            label={i18n.translate('indexPatternFieldEditor.fieldPreview.documentIdField.label', {
+              defaultMessage: 'Document ID',
+            })}
+            error={errorMessage}
+            isInvalid={isInvalid}
+            fullWidth
+          >
+            <EuiFieldText
+              isInvalid={isInvalid}
+              value={documentId}
+              onChange={onDocumentIdChange}
+              isLoading={isLoading}
+              fullWidth
+              data-test-subj="documentIdField"
+            />
+          </EuiFormRow>
+          {isCustomID && (
+            <span>
+              <EuiButtonEmpty color="primary" size="xs" flush="left" onClick={loadDocFromCluster}>
+                {i18n.translate(
+                  'indexPatternFieldEditor.fieldPreview.documentIdField.loadDocumentsFromCluster',
+                  {
+                    defaultMessage: 'Load documents from cluster',
+                  }
+                )}
+              </EuiButtonEmpty>
+            </span>
+          )}
+        </EuiFlexItem>
+
+        {showNavButtons && (
+          <EuiFlexItem grow={false}>
+            <EuiFlexGroup gutterSize="s" alignItems="flexEnd">
+              <EuiFlexItem grow={false}>
+                <EuiButtonIcon
+                  display="base"
+                  size="m"
+                  onClick={prev}
+                  iconType="arrowLeft"
+                  aria-label={i18n.translate(
+                    'indexPatternFieldEditor.fieldPreview.documentNav.previousArialabel',
+                    {
+                      defaultMessage: 'Previous',
```

Comment:
It's best to make `aria-label`s a little more descriptive to provide all the context necessary from the button itself. Previous what?

I'm not sure if this is correct but it might be something like
```suggestion
                      defaultMessage: 'Previous document',
```